db.go

```diff
@@ -423,17 +423,26 @@ func (db *DB) mmap(minsz int) error {
 	}
 
 	// Unmap existing data before continuing.
-	if err := db.munmap(); err != nil {
+	if err = db.munmap(); err != nil {
 		return err
 	}
 
 	// Memory-map the data file as a byte slice.
 	// gofail: var mapError string
 	// return errors.New(mapError)
-	if err := mmap(db, size); err != nil {
+	if err = mmap(db, size); err != nil {
 		return err
 	}
 
+	// Perform unmmap on any error to reset all data fields:
+	// dataref, data, datasz, meta0 and meta1.
+	defer func() {
+		if err != nil {
+			// TODO(ahrtr): add log to print the error if unmap somehow fails.
+			_ = db.munmap()
```

Comment:
Thanks @ahrtr, overall looks good! 

I'm wondering about the failure modes here reading through the man page: https://linux.die.net/man/2/munmap

it kinda looks to me that most of these errors should also be fatal, so I wouldn't even just log them but directly error out of the process. wdyt?

Reply:
or at the very least, wrap the already existing `err` with the new one